Display time it took the process to run in Verbose mode

perk11 opened the pull request:
To achieve this I also had to rework waiting for the last processes, so now the progressbar rendering in the end of the queue is done when every process is terminated, not just once when they all are. This fixes a weird behaviour in verbose log that in the end of the queue there was always `$totalParallelProcesses` less log lines then total, even if just one process is running.

perk11 commented:
This breaks php 5.3 support, if you want to still support it, ping me, I guess just removing `callable` typehint should solve this.

liuggio commented:
@perk11  there are few big changes, thanks a lot for this

liuggio commented:
@perk11  great job but there are some BCs, is possible to avoid those?

perk11 commented:
@liuggio If you meant PHP 5.3 compatibility break, I fixed that.

liuggio commented:
@perk11 i was talking about the breaking compatibility some public methods are renamed or deleted , all the public should be considered as API 
but in about 2 hours I'll go deeper in the code I'll update you ...

perk11 commented:
@liuggio I didn't think of it this way, only thought of use case where fastest is used as a binary. 

I changed 2 public methods: 
1. Removed `isAnyStillRunning`, since I didn't see it being used anywhere,this is easy to restore.
2. Altered `Report` class constructor to add `time`. This is possible to avoid with setter, but that would be ugly.

liuggio commented:
thanks @perk11 great job (again)
